### PR TITLE
ZCS-12397: WCAG | Fixed medium complexity Aria issues part-1

### PR DESCRIPTION
**Pull request by miteshsavani810**
See the changes of zm-web-client on this pull request #https://github.com/Zimbra/zm-web-client/pull/737

### CODE REVIEW

File: WebRoot/js/ajax/dwt/widgets/DwtTreeItem.js [hunk 1]
```diff
@@ -598,12 +598,13 @@ function(index, realizeDeferred, forceNode) {
 	this._extraCell = document.getElementById(data.id + "_extraCell");
 
 	/* assign the ARIA level */
-	this.setAttribute("aria-level", this.getNestingLevel());
+	this._itemDiv.setAttribute("aria-level", this.getNestingLevel());
+	this._itemDiv.setAttribute("role","heading");
```
Comment (silentsakky) [inline WebRoot/js/ajax/dwt/widgets/DwtTreeItem.js]:
```suggestion
	this._itemDiv.setAttribute("role", "heading");
```

File: WebRoot/js/ajax/dwt/widgets/DwtTreeItem.js [hunk 2]
```diff
@@ -598,12 +598,13 @@ function(index, realizeDeferred, forceNode) {
...
 
 	/* add a label for screenreaders, so that they don't read the entire
 	   element */
 	if (this._textCell) {
-		this.setAttribute("aria-labelledby", this._textCell.id);
+		this._itemDiv.setAttribute("aria-labelledby", this._textCell.id);
```
Comment (silentsakky) [inline WebRoot/js/ajax/dwt/widgets/DwtTreeItem.js]: should we check if _itemDiv exists?

File: WebRoot/js/ajax/dwt/widgets/DwtControl.js [hunk 3]
```diff
@@ -1139,13 +1143,22 @@ function(state) {
...
-        }
-    }).bind(this));
+	// Set Aria states.
+	var ariaStateKeys = Object.keys(DwtControl._ARIA_STATES);
+	var attribute = DwtControl._ARIA_STATES[state];
+
+	// Add/remove aria states only if given state is blank or given state is present in _ARIA_STATES
+	// state will be blank when user navigate to other app. e.g. User is on Mail and navigate to Calendar then state='' for Mail app object.
+	if (attribute || state === '') {
+		for (var i=0; i < ariaStateKeys.length; i++) {
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/ajax/dwt/widgets/DwtControl.js]: `var i = 0;`
Please check the indentation. The code that was replaced had a different indentation.

File: WebRoot/js/ajax/dwt/widgets/DwtTreeItem.js (hunk 1 above)
Comment (miteshsavani810) [reply WebRoot/js/ajax/dwt/widgets/DwtTreeItem.js]: Fixed.

File: WebRoot/js/ajax/dwt/widgets/DwtTreeItem.js (hunk 2 above)
Comment (miteshsavani810) [reply WebRoot/js/ajax/dwt/widgets/DwtTreeItem.js]: Fixed.

File: WebRoot/js/ajax/dwt/widgets/DwtControl.js (hunk 3 above)
Comment (miteshsavani810) [reply WebRoot/js/ajax/dwt/widgets/DwtControl.js]: Fixed.

**Review by HimanshuSojitraSynacor — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Review by lavegupta — APPROVED** [review]

**Review by lavegupta — APPROVED** [review]

**Review by HimanshuSojitraSynacor — APPROVED** [review]

**Pull request merged**